File: voxsight/src/main/java/edu/cit/capstone/voxsight/dto/OmrResponse.java
package edu.cit.capstone.voxsight.dto;

public record OmrResponse(boolean success, String url, String filename, String error) {
    public static OmrResponse ofSuccess(String url, String filename) {
        return new OmrResponse(true, url, filename, null);
    }

    public static OmrResponse ofError(String error) {
        return new OmrResponse(false, null, null, error);
    }
}

File: mobile/app/src/main/java/com/cit/kaido/voxsight/network/OmrResponse.kt
package com.cit.kaido.voxsight.network

import com.google.gson.annotations.SerializedName

data class OmrResponse(
    @SerializedName("success") val success: Boolean = false,
    @SerializedName("url") val url: String?,
    @SerializedName("filename") val filename: String?,
    @SerializedName("error") val error: String?
)
